### DISCUSSION TITLE
Calling `setup_dap_main_class_configs()` before debugger starts

### DISCUSSION THREAD

**Discussion by paulkre** [mfussenegger/nvim-jdtls#592] [Q&A] [answered]
Is it possible to automatically run `require('jdtls.dap').setup_dap_main_class_configs()` before `require('dap').continue()` is executed? Currently I have to remember to call `require('jdtls.dap').setup_dap_main_class_configs()` before starting the debugger with `require('dap').continue()`. Where in the configuration is the best place to call `require('jdtls.dap').setup_dap_main_class_configs()`?

**Comment by mfussenegger** [comment, marked as answer]
You could chain the two:

```lua
require('jdtls.dap').setup_dap_main_class_configs({ on_ready = function() require("dap").continue() end })
```

**Reply by paulkre** [reply]
That worked perfectly, thanks!
